File: app/src/main/java/soup/movie/ui/main/MainViewState.java
package soup.movie.ui.main;

import soup.movie.ui.BaseViewState;

interface MainViewState extends BaseViewState {

    class NowState implements MainViewState {

        private final boolean isVerticalType;

        NowState(boolean verticalType) {
            isVerticalType = verticalType;
        }

        public boolean isVerticalType() {
            return isVerticalType;
        }

        @Override
        public String toString() {
            return "NowState{" +
                    "isVerticalType=" + isVerticalType +
                    '}';
        }
    }

    class PlanState implements MainViewState {

        private final boolean isVerticalType;

        PlanState(boolean verticalType) {
            isVerticalType = verticalType;
        }

        public boolean isVerticalType() {
            return isVerticalType;
        }

        @Override
        public String toString() {
            return "PlanState{" +
                    "isVerticalType=" + isVerticalType +
                    '}';
        }
    }

    class SettingsState implements MainViewState {
        SettingsState() {
        }

        @Override
        public String toString() {
            return "SettingsState{}";
        }
    }
}

File: app/src/main/java/soup/movie/ui/main/MainViewState.kt
package soup.movie.ui.main

import androidx.annotation.Keep

sealed class MainViewState {

    @Keep
    object NowState : MainViewState() {

        override fun toString(): String = javaClass.simpleName
    }

    @Keep
    object PlanState : MainViewState() {

        override fun toString(): String = javaClass.simpleName
    }

    @Keep
    object TheatersState : MainViewState() {

        override fun toString(): String = javaClass.simpleName
    }

    @Keep
    object SettingsState : MainViewState() {

        override fun toString(): String = javaClass.simpleName
    }
}
